### ISSUE TITLE
az webapp up ZIP deployment fails

### ISSUE THREAD

**Issue by dnj55** [Azure/azure-cli#11261]
Per guidance from your team concerning a known issue with az webapp up, I downgraded to 2.0.75. However, the ZIP deployment of my webapp fails (although checking in my Azure portal, the app service is successfully created).

Thank you.

**Command Name**
`az webapp up`

**Errors:**
```
Zip deployment failed. {'id': '9aa4f222956f41579af84b2832bff1be', 'status': 3, 'status_text': '', 'author_email': 'N/A', 'author': 'N/A', 'deployer': 'Push-Deployer', 'message': 'Created via a push deployment', 'progress': '', 'received_time': '2019-11-16T03:22:35.7796793Z', 'start_time': '2019-11-16T03:22:35.9229355Z', 'end_time': '2019-11-16T03:22:36.3614476Z', 'last_success_end_time': None, 'complete': True, 'active': False, 'is_temp': False, 'is_readonly': True, 'url': 'https://kaboom201911152.scm.azurewebsites.net/api/deployments/latest', 'log_url': 'https://kaboom201911152.scm.azurewebsites.net/api/deployments/latest/log', 'site_name': 'kaboom201911152'}. Please run the command az webapp log tail
                           -n kaboom201911152 -g contoso-travel-rg
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az webapp up -n {} --resource-group {} --location {}`

## Environment Summary
```
Windows-10-10.0.17763-SP0
Python 3.6.6
Shell: cmd.exe

azure-cli 2.0.71 *

```

**Comment by panchagnula** [comment]
can you share the error you received that states that the zip deployment failed?

**Comment by dnj55** [comment]
Sure thing. Attached.

**Comment by panchagnula** [comment]
@dnj55 sorry I still don't see the error messages can you just copy the error stack trace or message please?

**Comment by dnj55** [comment]
Sure thing. Here's the error message I get:

Starting zip deployment. This operation can take a while to complete ...
Configuring default logging for the app, if not already enabled
Zip deployment failed. {‘id': '9aa4f222956f41579af84b2832bfflbe', ‘status’:
3, ‘status_text': '', ‘author_email': ‘N/A’, ‘author’: 'N/A', ‘deployer':
'Push-Deployer', ‘message’: ‘Created via a push deployment', ‘progress’:
'', ‘received_time': '2019-11-16T03:22:35.7796793Z', ‘start_time’:
'2019-11-16T03:22:35.9229355Z', ‘end_time': '2019-11-16T@3:22:36.3614476Z',
'last_success_end_time': None, 'complete': True, 'active': False,
'is_temp': False, 'is_readonly': True, 'url':'
https://kaboom201911152.scm.azurewebsites.net/api/deployments/latest',
'log_url': 'https://kaboom201911152,scm,
azurewebsites.net/api/deployments/latest/log', 'site_name':
'kabom201911152'}. Please run the command az webapp log tail -n
kaboom201911152 -g contoso-travel-rg

**Comment by panchagnula** [comment]
Could you please re-try let us know if you see the same issue with the latest version of CLI 

Also, if this still fails can you run the az webapp log tail command that is suggested in the error message & share the details?

**Comment by panchagnula** [comment]
closing due to lack of activity - please reactivate if you are still seeing this with the latest version of CLI. Thanks!

**Comment by phuot** [comment]
I have a very similar comment when trying to use az webapp deployment:
az webapp deployment source config-zip --resource-group $resourceGroupName --name $WebAppName --src $src --slot $slotName

It happen most of the time when deploying to a slot that has been created few seconds ago.

here is my creation workflow (in case other steps might be causing the issue):
```
az webapp deployment slot create --name $webAppName --resource-group $resourceGroupName --slot $slotName --configuration-source $webAppName

az webapp identity assign --resource-group $resourceGroupName --name $webAppName --slot $slotName --query principalId --output tsv

az webapp config appsettings set --resource-group $resourceGroupName --name $webAppName --slot $slotName --settings APPINSIGHTS_INSTRUMENTATIONKEY=$appInsightKey WEBSITE_LOAD_CERTIFICATES=*

az webapp deployment source config-zip --resource-group $resourceGroupName --name $webAppName --src $src --slot $slotName
```

note: I am using azure-cli V2.0.77

sometime the error is similar to :
```
az : WARNING: Getting scm site credentials for zip deployment
At line:79 char:15
+ ...   $result = az webapp deployment source config-zip --resource-group $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (WARNING: Gettin... zip deployment:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
WARNING: Starting zip deployment. This operation can take a while to complete ...
WARNING: Deployment endpoint responses with status code 503
```

sometime we receive (replaced sensitive info with *****):
```
az : WARNING: Getting scm site credentials for zip deployment
At line:79 char:15
+ ...   $result = az webapp deployment source config-zip --resource-group $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (WARNING: Gettin... zip deployment:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
WARNING: Starting zip deployment. This operation can take a while to complete ...
WARNING: Deployment endpoint responses with status code 202
WARNING: Configuring default logging for the app, if not already enabled
ERROR: Zip deployment failed. {'id': 'temp-8f69cfd5', 'status': 3, 'status_text': '', 'author_email': 'N/A', 'author': 'N/A', 'deployer': 'Push-Deployer', 'message': 'Deploying from pushed zip file', 'progress': '', 'received_time': 
'2020-02-10T19:12:51.4052348Z', 'start_time': '2020-02-10T19:12:51.4052348Z', 'end_time': '2020-02-10T19:12:58.7646024Z', 'last_success_end_time': None, 'complete': True, 'active': False, 'is_temp': True, 'is_readonly': False, 'url': 
'https://*********-test.scm.azurewebsites.net/api/deployments/latest', 'log_url': 'https://*********-test.scm.azurewebsites.net/api/deployments/latest/log', 'site_name': '*********', 'provisioningState': None}. 
Please run the command az webapp log tail
                           -n *********-g *********
```

If I look at the KUDU logs (https://*********-test.scm.azurewebsites.net/api/deployments/latest/log), I see the following error:
```
{
Message: "An error has occurred.",
ExceptionMessage: "No log found for 'latest'.",
ExceptionType: "System.IO.FileNotFoundException",
StackTrace: " at Kudu.Core.Deployment.DeploymentManager.GetLogEntries(String id) in C:\Kudu Files\Private\src\master\Kudu.Core\Deployment\DeploymentManager.cs:line 94 at Kudu.Services.Deployment.DeploymentController.GetLogEntry(String id) in C:\Kudu Files\Private\src\master\Kudu.Services\Deployment\DeploymentController.cs:line 375"
}
```

submitted the issue https://github.com/Azure/azure-cli/issues/12125

**Comment by jongio** [comment]
Seeing this today:


.NET Core SDK (3.1.101)

azure-cli                          2.2.0

command-modules-nspkg              2.0.3
core                               2.2.0
nspkg                              3.0.4
telemetry                          1.0.4

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\JonGallant\.azure\cliextensions'

Python (Windows) 3.6.6 (v3.6.6:4cf1f54eb7, Jun 27 2018, 02:47:15) [MSC v.1900 32 bit (Intel)]

Commands:

```
dotnet new webapi --no-https
az webapp up -n azuptest -g jongrg
```

```
webapp azuptest doesn't exist
Creating webapp 'azuptest' ...
Configuring default logging for the app, if not already enabled
Creating zip with contents of dir C:\temp\azuptest ...
Getting scm site credentials for zip deployment
Starting zip deployment. This operation can take a while to complete ...
Deployment endpoint responded with status code 202
Configuring default logging for the app, if not already enabled
Zip deployment failed. {'id': '9054b7501d6d479f8f466a1e0183320f', 'status': 3, 'status_text': '', 'author_email': 'N/A', 'author': 'N/A', 'deployer': 'Push-Deployer', 'message': 'Created via a push deployment', 'progress': '', 'received_time': '2020-03-16T22:57:06.5245708Z', 'start_time': '2020-03-16T22:57:06.6339499Z', 'end_time': '2020-03-16T22:57:14.0641852Z', 'last_success_end_time': None, 'complete': True, 'active': False, 'is_temp': False, 'is_readonly': True, 'url': 'https://azuptest.scm.azurewebsites.net/api/deployments/latest', 'log_url': 'https://azuptest.scm.azurewebsites.net/api/deployments/latest/log', 'site_name': 'azuptest', 'provisioningState': None}. Please run the command az webapp log tail
                           -n azuptest -g jongrg
```

**Comment by panchagnula** [comment]
Thanks for sharing from our logs I do see an error on Kudu while completing zip deployment  - following up with them to see if this any known issue with Kudu.

**Comment by panchagnula** [comment]
As per our Kudu folks the error is because the dotnet source code is trying to use dotnet core 3.1 SDK that is still not available on these VMs- since the deployment for this is still in progress and the changes are rolling out. For now if you want to use this SDK the work around would be to not use webapp up -> but do 
dotnet new webapi --no-https
dotnet restore
dotnet build 
Create a Zip file and use
az webapp deployment source config-zip

Once our deployment is complete you should be able to use az webapp up successfully.

**Comment by josh-qf** [comment]
Hi @panchagnula. I'm running into the same issue with a Python app deploy and tried the `config-zip` command with the same failure:
```Zip deployment failed. {'id': '', 'status': 3, 'status_text': '', 'author_email': 'N/A', 'author': 'N/A', 'deployer': 'Push-Deployer', 'message': 'Created via a push deployment', 'progress': '', 'received_time': '2020-03-19T05:09:38.8865191Z', 'start_time': '2020-03-19T05:09:39.0396155Z', 'end_time': '2020-03-19T05:10:14.5460089Z', 'last_success_end_time': None, 'complete': True, 'active': False, 'is_temp': False, 'is_readonly': True, ...}. Please run the command az webapp log tail```

When trying to access the log, I get this error:
```{"ClassName":"System.IO.FileNotFoundException","Message":"No log found for 'latest'.","Data":null,"InnerException":null,"HelpURL":null,"StackTraceString":"   at Kudu.Core.Deployment.DeploymentManager.GetLogEntries(String id) in /tmp/KuduLite/Kudu.Core/Deployment/DeploymentManager.cs:line 111\n   at Kudu.Services.Deployment.DeploymentController.GetLogEntry(String id) in /tmp/KuduLite/Kudu.Services/Deployment/DeploymentController.cs:line 432","RemoteStackTraceString":null,"RemoteStackIndex":0,"ExceptionMethod":null,"HResult":-2147024894,"Source":"Kudu.Core","WatsonBuckets":null,"FileNotFound_FileName":null,"FileNotFound_FusionLog":null}```

Please assist, thanks!

**Comment by kausali** [comment]
Hi @panchagnula I am running into similar issue trying to deploy a bot that use Dotnet core sdk. 
When I runt this command, I get Zip deployment failed error :
 `az webapp deployment source config-zip --resource-group "Test-Bot" --name "Test-Application" --src "Test.zip"` 

If I look at the log_url, I see this message:

`{"Message":"An error has occurred.","ExceptionMessage":"No log found for 'latest'.","ExceptionType":"System.IO.FileNotFoundException","StackTrace":"   at Kudu.Core.Deployment.DeploymentManager.GetLogEntries(String id) in C:\\Kudu Files\\Private\\src\\master\\Kudu.Core\\Deployment\\DeploymentManager.cs:line 98\r\n   at Kudu.Services.Deployment.DeploymentController.GetLogEntry(String id) in C:\\Kudu Files\\Private\\src\\master\\Kudu.Services\\Deployment\\DeploymentController.cs:line 375"}`

I followed the workaround you mentioned above but I still get the same error. 
I have Azure CLI - 2.2.0

**Comment by panchagnula** [comment]
@kausali this is not the same error as above - the file not found exception is coming since the command is not able to find the Test.zip file - try running the above command by giving the full path of the test.zip file. Thanks!

**Comment by kausali** [comment]
I am in the same directory as where the .zip folder is located. I have also tried from a different location and giving a full path of the zip folder.  
![image](https://user-images.githubusercontent.com/27708074/77096031-f9e47d00-69cb-11ea-87d8-54ac6d6c0dea.png)

**Comment by panchagnula** [comment]
@kausali  this is a different error - so I think earlier the path was incorrect. Did you happen to check the scm site(link is in the error stack trace) for logs? Unfortunately I cannot be of help without the details on appName - If you have the Kudu error message you can file an issue on Kudu directly for this. Thanks!

**Comment by mgovers-brightcubes** [comment]
I have the same issue when following the exercises in this learning path: https://docs.microsoft.com/en-us/learn/modules/host-a-web-app-with-azure-app-service/ . I don't know if the location has anything to do with the issue, but I am trying to access the Central US location (since that is the one in the Sandbox environment) from the Netherlands.

**Comment by panchagnula** [comment]
closing this since the original issue was fixed. Please create new issue if some one is still seeing this with the latest CLI version. Thanks!

**Comment by chinmaymundhada** [comment]
Hello I received the error as given below during deployment
Deployment endpoint responded with status code 202
Configuring default logging for the app, if not already enabled
Zip deployment failed. {'id': 'temp-2139d583', 'status': 3, 'status_text': '', 'author_email': 'N/A', 'author': 'N/A', 'deployer': 'ZipDeploy', 'message': 'Deploying from pushed zip file', 'progress': '', 'received_time': '2020-08-09T13:40:28.1936884Z', 'start_time': '2020-08-09T13:40:28.1936884Z', 'end_time': '2020-08-09T13:40:55.0011174Z', 'last_success_end_time': None, 'complete': True, 'active': False, 'is_temp': True, 'is_readonly': False, 'url': 'https://tejgyanchatbot1.scm.azurewebsites.net/api/deployments/latest', 'log_url': 'https://tejgyanchatbot1.scm.azurewebsites.net/api/deployments/latest/log', 'site_name': 'tejgyanchatbot1', 'provisioningState': None}. Please run the command az webapp log deployment show
                           -n tejgyanchatbot1 -g ChatbotTejgyan2

**Comment by Aaronmcd1978** [comment]
Hi I am trying to do a zip deployment from my download folder in CLI and i get the following error 
Zip deployment failed. {'id': '206d94e91e7049a48748061a41de549c', 'status': 3, 'status_text': '', 'author_email': 'N/A', 'author': 'N/A', 'deployer': 'ZipDeploy', 'message': 'Created via a push deployment', 'progress': '', 'received_time': '2020-09-15T15:43:58.1498639Z', 'start_time': '2020-09-15T15:43:58.321764Z', 'end_time': '2020-09-15T15:44:02.9310477Z', 'last_success_end_time': None, 'complete': True, 'active': False, 'is_temp': False, 'is_readonly': True, 'url': 'https://logziolfunctionapp3ntx6o26nxd5g.scm.azurewebsites.net/api/deployments/latest', 'log_url': 'https://logziolfunctionapp3ntx6o26nxd5g.scm.azurewebsites.net/api/deployments/latest/log', 'site_name': 'logzioLFunctionApp3ntx6o26nxd5g', 'provisioningState': None}. Please run the command az webapp log deployment show
                           -n logzioLFunctionApp3ntx6o26nxd5g -g App_test

**Comment by roalexan** [comment]
For what it's worth, I kept getting this error until I removed several extraneous files that I had put into the app folder. I'm especially suspicious of some zips I had in there.

**Comment by xostap** [comment]
> I have a very similar comment when trying to use az webapp deployment:
> az webapp deployment source config-zip --resource-group $resourceGroupName --name $WebAppName --src $src --slot $slotName
> 
> It happen most of the time when deploying to a slot that has been created few seconds ago.
> 
> here is my creation workflow (in case other steps might be causing the issue):
> 
> ```
> az webapp deployment slot create --name $webAppName --resource-group $resourceGroupName --slot $slotName --configuration-source $webAppName
> 
> az webapp identity assign --resource-group $resourceGroupName --name $webAppName --slot $slotName --query principalId --output tsv
> 
> az webapp config appsettings set --resource-group $resourceGroupName --name $webAppName --slot $slotName --settings APPINSIGHTS_INSTRUMENTATIONKEY=$appInsightKey WEBSITE_LOAD_CERTIFICATES=*
> 
> az webapp deployment source config-zip --resource-group $resourceGroupName --name $webAppName --src $src --slot $slotName
> ```
> 
> note: I am using azure-cli V2.0.77
> 
> sometime the error is similar to :
> 
> ```
> az : WARNING: Getting scm site credentials for zip deployment
> At line:79 char:15
> + ...   $result = az webapp deployment source config-zip --resource-group $ ...
> +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : NotSpecified: (WARNING: Gettin... zip deployment:String) [], RemoteException
>     + FullyQualifiedErrorId : NativeCommandError
>  
> WARNING: Starting zip deployment. This operation can take a while to complete ...
> WARNING: Deployment endpoint responses with status code 503
> ```
> 
> sometime we receive (replaced sensitive info with *****):
> 
> ```
> az : WARNING: Getting scm site credentials for zip deployment
> At line:79 char:15
> + ...   $result = az webapp deployment source config-zip --resource-group $ ...
> +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : NotSpecified: (WARNING: Gettin... zip deployment:String) [], RemoteException
>     + FullyQualifiedErrorId : NativeCommandError
>  
> WARNING: Starting zip deployment. This operation can take a while to complete ...
> WARNING: Deployment endpoint responses with status code 202
> WARNING: Configuring default logging for the app, if not already enabled
> ERROR: Zip deployment failed. {'id': 'temp-8f69cfd5', 'status': 3, 'status_text': '', 'author_email': 'N/A', 'author': 'N/A', 'deployer': 'Push-Deployer', 'message': 'Deploying from pushed zip file', 'progress': '', 'received_time': 
> '2020-02-10T19:12:51.4052348Z', 'start_time': '2020-02-10T19:12:51.4052348Z', 'end_time': '2020-02-10T19:12:58.7646024Z', 'last_success_end_time': None, 'complete': True, 'active': False, 'is_temp': True, 'is_readonly': False, 'url': 
> 'https://*********-test.scm.azurewebsites.net/api/deployments/latest', 'log_url': 'https://*********-test.scm.azurewebsites.net/api/deployments/latest/log', 'site_name': '*********', 'provisioningState': None}. 
> Please run the command az webapp log tail
>                            -n *********-g *********
> ```
> 
> If I look at the KUDU logs (https://*********-test.scm.azurewebsites.net/api/deployments/latest/log), I see the following error:
> 
> ```
> {
> Message: "An error has occurred.",
> ExceptionMessage: "No log found for 'latest'.",
> ExceptionType: "System.IO.FileNotFoundException",
> StackTrace: " at Kudu.Core.Deployment.DeploymentManager.GetLogEntries(String id) in C:\Kudu Files\Private\src\master\Kudu.Core\Deployment\DeploymentManager.cs:line 94 at Kudu.Services.Deployment.DeploymentController.GetLogEntry(String id) in C:\Kudu Files\Private\src\master\Kudu.Services\Deployment\DeploymentController.cs:line 375"
> }
> ```
> 
> submitted the issue #12125

+1

**Comment by Rockyparkour** [comment]
> I am in the same directory as where the .zip folder is located. I have also tried from a different location and giving a full path of the zip folder.
> ![image](https://user-images.githubusercontent.com/27708074/77096031-f9e47d00-69cb-11ea-87d8-54ac6d6c0dea.png)

Did you find any solution till now? I am facing the same.

**Comment by sfmskywalker** [comment]
Same problem here:

![image](https://user-images.githubusercontent.com/938393/111512168-c0199400-874f-11eb-8b9a-b24a3c6a42aa.png)

**Comment by iddqdiddqd** [comment]
> > I am in the same directory as where the .zip folder is located. I have also tried from a different location and giving a full path of the zip folder.
> > ![image](https://user-images.githubusercontent.com/27708074/77096031-f9e47d00-69cb-11ea-87d8-54ac6d6c0dea.png)
> 
> Did you find any solution till now? I am facing the same.

I was having the same error. It was resolved by re-adding the files to a new zip archive.

Initially, I did `tar -czf file.zip *` and az webapp deploy was failing with the error from your screenshot.

After executing `zip file.zip *`, the deploy went successfully.

**Comment by panchagnula** [comment]
@iddqdiddqd & all I am sorry for the experience with zip deploy & all of you helping each other with suggestions.
The error indicates a run time issue with Zip deploy itself, & not the command specific problem. Ideally command & CLI would do a better job surfacing the errors - we are working the team that owns deployments, to surface these logs better & also make deployment handle zip files created by various tools (tar as well). While we don't have an ETA yet, please expect some improvements next release starting with changes to az webapp deploy command https://docs.microsoft.com/en-us/cli/azure/webapp?view=azure-cli-latest#az-webapp-deploy, which uses a New API for deployment.  @StrawnSC FYI

**Comment by Nikostito** [comment]
I am facing the exact same issue to one of my workflows. Yesterday the `az webapp up` command worked fine.
Today it gives me the same zip error. Should we move forward to `az webapp deploy` judging from your comment above?
Please let me know.

**Comment by panchagnula** [comment]
> I am facing the exact same issue to one of my workflows. Yesterday the `az webapp up` command worked fine. Today it gives me the same zip error. Should we move forward to `az webapp deploy` judging from your comment above? Please let me know.
> 
> Edit: Here's a log I maanged to get from Azure VsCode Plugin (Az Portal shows no logs whatsoever) It seems to me that zip file gets uploaded correctly but then yarn install fails. @panchagnula
> 
> `2022-03-22T13:55:31.995Z - Running 'yarn install --prefer-offline'... 2022-03-22T13:55:32.207Z - yarn install v1.22.10 2022-03-22T13:55:32.345Z - warning package.json: No license field 2022-03-22T13:55:32.394Z - warning No license field 2022-03-22T13:55:32.403Z - [1/4] Resolving packages... 2022-03-22T13:55:32.837Z - [2/4] Fetching packages... 2022-03-22T13:56:07.040Z - info There appears to be trouble with your network connection. Retrying... 2022-03-22T13:56:40.197Z - info There appears to be trouble with your network connection. Retrying... 2022-03-22T13:57:04.141Z - info There appears to be trouble with your network connection. Retrying... 2022-03-22T13:57:13.324Z - info There appears to be trouble with your network connection. Retrying... 2022-03-22T13:57:46.439Z - info There appears to be trouble with your network connection. Retrying... 2022-03-22T13:58:00.837Z - info There appears to be trouble with your network connection. Retrying... 2022-03-22T13:58:12.810Z - info There appears to be trouble with your network connection. Retrying... 2022-03-22T13:58:33.745Z - error An unexpected error occurred: "https://registry.yarnpkg.com/date-fns/-/date-fns-2.28.0.tgz: ESOCKETTIMEDOUT". 2022-03-22T13:58:33.760Z - info If you think this is a bug, please open a bug report with the information provided in "/tmp/8da0c0b9f0223ca/yarn-error.log". 2022-03-22T13:58:33.789Z - info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command. 2022-03-22T13:58:34.027Z - info There appears to be trouble with your network connection. Retrying... 2022-03-22T13:59:28.608Z - info There appears to be trouble with your network connection. Retrying... 2022-03-22T14:01:01.529Z - warning package.json: No license field\nwarning No license field\nerror An unexpected error occurred: "https://registry.yarnpkg.com/date-fns/-/date-fns-2.28.0.tgz: ESOCKETTIMEDOUT".\n/opt/Kudu/Scripts/starter.sh oryx build /tmp/zipdeploy/extracted -o /home/site/wwwroot --platform nodejs --platform-version 16 -i /tmp/8da0c0b9f0223ca -p compress_node_modules=tar-gz --log-file /tmp/build-debug.log 2022-03-22T14:01:02.097Z - Deployment Failed.`

@Nikostito this seems like a runtime failure on Kudu end, when trying to build some packages. Have you made any changes on your end (add private endpoint or new contents on the zip file?)  CLI command has some changes planned to surface deployment errors - but the error itself is something that needs assistance from the feature team.

**Comment by robertmuehsig** [comment]
I had the same problem and the source was, that the `c:\local` storage was full:

![image](https://github.com/Azure/azure-cli/assets/756703/d52a5dae-1004-479b-90f4-c8b53e1a5e93)

We have multiple apps on the same app plan and zip deploy leaves a `c:\local\temp\zipdeploy` folder after each deployment (per app) and if have enough apps or your deployment zip is big enough, you will run out of space.

Solution: Delete the temp zipdeploy folder for each app (the space is shared, but each app has it's own drive).
